dask/array/ghost.py

```diff
@@ -258,28 +250,45 @@ def boundaries(x, depth=None, kind=None):
     periodic
     constant
     """
-    if not isinstance(kind, dict):
-        kind = dict((i, kind) for i in range(x.ndim))
-    if not isinstance(depth, dict):
-        depth = dict((i, depth) for i in range(x.ndim))
+    depth = coerce_depth(x.ndim, depth)
+    kind = coerce_boundary(x.ndim, kind)
 
     for i in range(x.ndim):
         d = depth.get(i, 0)
         if d == 0:
             continue
 
-        if kind.get(i) == 'periodic':
-            x = periodic(x, i, d)
-        elif kind.get(i) == 'reflect':
-            x = reflect(x, i, d)
-        elif kind.get(i) == 'nearest':
-            x = nearest(x, i, d)
-        elif i in kind:
-            x = constant(x, i, d, kind[i])
-
+        left_kind, right_kind = kind.get(i, ('reflect', 'reflect'))
+        l = handed_boundary('left', x, i, d, left_kind)
+        r = handed_boundary('right', x, i, d, right_kind)
+        if (l is None) and (r is None):
+            continue
+        elif l is None:
+            x = concatenate([x, r], axis=i)
+        elif r is None:
+            x = concatenate([l, x], axis=i)
+        else:
+            x = concatenate([l, x, r], axis=i)
     return x
 
 
+def handed_boundary(hand, x, axis, depth, kind):
+    if kind is None:
+        return None
+    elif kind == 'periodic':
+        b = periodic(hand, x, axis, depth)
+    elif kind == 'reflect':
+        b = reflect(hand, x, axis, depth)
+    elif kind == 'nearest':
+        b = nearest(hand, x, axis, depth)
+    elif isinstance(kind, (float, int)): #XXX: check for scalars better
```

Comment:
I need to check for scalars in a better way here. This would fail for a numpy scalar.